Sources/AWSLambdaEvents/Cloudwatch.swift

```diff
@@ -131,7 +132,7 @@ public enum CloudwatchDetails {
 
     struct TypeMismatch: Error {
         let name: String
-        let type: Any
+        let type: any Sendable
```

Comment:
I tried to stay close to the original code (`Any`)

The change you suggested generates compilation errors 
```
~/code/swift/lambda/swift-aws-lambda-events/Sources/AWSLambdaEvents/Cloudwatch.swift:135:13: error: property declares an opaque return type, but has no initializer expression from which to infer an underlying type
        let type: some CloudwatchDetail
```

Reply:
`any` should work

Reply:
When I tried any, I received a really weird error 
```
/Users/stormacq/Documents/amazon/code/swift/lambda/swift-aws-lambda-events/Sources/AWSLambdaEvents/Cloudwatch.swift:73:74: error: argument type 'Detail.Type' does not conform to expected type 'CloudwatchDetail'
            throw CloudwatchDetails.TypeMismatch(name: detailType, type: Detail.self)
```

I also tried 

```
            throw CloudwatchDetails.TypeMismatch(name: detailType, type: Detail.self as! CloudwatchDetail)

```

Which is strange because `Detail` is the name of the generic and it is declared as `CloudwatchDetail` 

But this breaks the test https://github.com/swift-server/swift-aws-lambda-events/blob/main/Tests/AWSLambdaEventsTests/CloudwatchTests.swift#L129

Do you think this case can happen outside a test ?

Reply:
OK, I gave it a fresh look this morning and found the solution.